scripts/dataset/export/index.js

```diff
@@ -0,0 +1,124 @@
+  const versionsStorageAdapter = await initVersionsStorageAdapter();
+  const { archive, archiveName, archiveWrittenPromise } = await initializeArchive(archivePath);
+
+  await renamer.loadRules();
+
+  const periodDates = { first: null, last: null };
+  const services = new Set();
+
+  for await (const record of versionsStorageAdapter.iterate()) {
+    const { serviceId, documentType, ...othersRecordAttributes } = record;
+    const { serviceId: renamedServiceId, documentType: renamedDocumentType } = renamer.applyRules(serviceId, documentType);
+
+    addRecordToArchive({
+      record: { serviceId: renamedServiceId, documentType: renamedDocumentType, ...othersRecordAttributes },
+      periodDates,
+      services,
+      archive,
+      archiveName,
+    });
+  }
+
+  addReadmeAndLicenseToArchive({ releaseDate, periodDates, services, archive, archiveName });
+
+  archive.finalize();
+
+  await archiveWrittenPromise;
+
+  return {
+    servicesCount: services.size,
+    firstCommitDate: periodDates.first.toLocaleDateString(locale, dateOptions),
+    lastCommitDate: periodDates.last.toLocaleDateString(locale, dateOptions),
+  };
+}
+
+async function initVersionsStorageAdapter() {
+  let versionsStorageAdapter;
+
+  if (config.has('recorder.versions.storage.git')) {
+    versionsStorageAdapter = new GitAdapter({
+      ...config.get('recorder.versions.storage.git'),
+      path: path.resolve(__dirname, '../../../', config.get('recorder.versions.storage.git.path')),
+      fileExtension: 'md',
+    });
+  } else if (config.has('recorder.versions.storage.mongo')) {
+    versionsStorageAdapter = new MongoAdapter(config.get('recorder.versions.storage.mongo'));
+  }
+
+  await versionsStorageAdapter.initialize();
+
+  return versionsStorageAdapter;
+}
+
+async function initializeArchive(archivePath) {
+  fsApi.mkdirSync(path.dirname(archivePath), { recursive: true });
+
+  const archiveExtension = path.extname(archivePath);
+  const archiveName = path.basename(archivePath, archiveExtension);
+
+  const output = fsApi.createWriteStream(archivePath);
+  const archive = archiver('zip', { zlib: { level: 9 } });
+
+  const archiveWrittenPromise = new Promise(resolve => {
+    output.on('close', () => resolve());
+  });
+
+  archive.pipe(output);
+
+  return { archive, archiveName, archiveWrittenPromise };
+}
+
+function addRecordToArchive({ record, periodDates, services, archive, archiveName }) {
+  const { serviceId, documentType, content, fetchDate } = record;
+
+  if (!periodDates.last || new Date(fetchDate) > new Date(periodDates.last)) {
+    periodDates.last = fetchDate;
+  }
+
+  if (!periodDates.first || new Date(fetchDate) < new Date(periodDates.first)) {
+    periodDates.first = fetchDate;
+  }
+
+  services.add(serviceId);
+
+  const fileName = fetchDate.toISOString()
+    .replace(/\.\d{3}/, '') // Remove milliseconds
+    .replace(/:|\./g, '-'); // Replace `:` and `.` by `-` to be compliant with the file system
```

Comment:
Why adding space and why removing caps. What is your rules to choose if we use a caps or not when commenting?

Reply:
Sorry, indeed this should be linted if we reach an agreement 🙂

I personally use the following rule:

- Multi-line comments start with `/**`, end with a line containing only `*/` and are full sentences, with capitals and punctuation.
- Single line comments are always at the end of the code section they comment (no single line comment on its own) to avoid rot, are separated from the code by two spaces, and have no capital nor punctuation. This helps me feel when my single-line comment is getting too long: if it feels like it should be two sentences, I should either refactor, improve the doc, or make that a multi-line comment.

Reply:
I tried `daysjs` but the format we need is not already defined and at the end we still have to do manual formatting. So I prefer avoid adding a dependency in this case.